### PR TITLE
feat(sentry apps): Validate black and white icon

### PR DESCRIPTION
**Pull request by ceorourke**
Ensure that if a user is uploading an image meant to be used for issue linking, that it is indeed black and white by checking the RBGA values.

### CODE REVIEW

File: src/sentry/api/bases/avatar.py [hunk 1]
```diff
@@ -30,12 +31,36 @@ class SentryAppLogoSerializer(serializers.Serializer):
     avatar_type = serializers.ChoiceField(choices=(("default", "default"), ("upload", "upload")))
     color = serializers.BooleanField(required=True)
 
+    def is_black_and_white(self, data):
```
Comment (ceorourke) [inline src/sentry/api/bases/avatar.py:34]: I know a triple nested for loop is not great but it's limited - `AvatarField` already checked the file size (and that it's a square image).

File: tests/sentry/api/endpoints/test_sentry_app_avatar.py [hunk 2]
```diff
@@ -20,11 +20,16 @@ def get_avatar(self, resp, is_color=True):
                 return avatar
 
     def create_avatar(self, is_color):
+        avatar_photo = (
+            b64encode(self.load_fixture("avatar.jpg"))
```
Comment (evanpurkhiser) [inline tests/sentry/api/endpoints/test_sentry_app_avatar.py:24]: Can we use a `png` fixture? We shouldn't allow jpg upload right?

File: src/sentry/api/bases/avatar.py (hunk 1 above)
Comment (mgaeta) [reply src/sentry/api/bases/avatar.py:34]: You should move this function to `/src/sentry/utils/avatar/`.

File: src/sentry/utils/avatar.py [hunk 3]
```diff
@@ -125,3 +126,12 @@ def get_email_avatar(
                     gravatar_url = get_gravatar_url(identifier, size=size)
                     return f'<img class="avatar" src="{gravatar_url}">'
     return get_letter_avatar(display_name, identifier, size, use_svg=False)
+
+
+def is_black_alpha_only(data: Any) -> bool:
+    """Check if an image has only black pixels"""
```
Comment (evanpurkhiser) [inline src/sentry/utils/avatar.py:132]:
```suggestion
    """Check if an image has only black pixels (with alpha)"""
```

File: src/sentry/api/fields/avatar.py [hunk 4]
```diff
@@ -45,6 +48,9 @@ def to_internal_value(self, data):
             raise ImageTooLarge()
 
         with Image.open(BytesIO(data)) as img:
+            if self.is_sentry_app and Image.MIME[img.format] not in SENTRY_APP_ALLOWED_MIMETYPES:
+                raise serializers.ValidationError("Invalid image format. Must use pngs.")
```
Comment (evanpurkhiser) [inline src/sentry/api/fields/avatar.py:52]:
```suggestion
							  valid_formats = ", ".join(SENTRY_APP_ALLOWED_MIMETYPES)
                raise serializers.ValidationError("Invalid image format. App icons should be {}.".format(valid_formats))
```

File: src/sentry/utils/avatar.py [hunk 5]
```diff
@@ -125,3 +126,12 @@ def get_email_avatar(
                     gravatar_url = get_gravatar_url(identifier, size=size)
                     return f'<img class="avatar" src="{gravatar_url}">'
     return get_letter_avatar(display_name, identifier, size, use_svg=False)
+
+
+def is_black_alpha_only(data: Any) -> bool:
```
Comment (evanpurkhiser) [inline src/sentry/utils/avatar.py:131]: Is there not a better type we can use here?

Comment (ceorourke) [reply src/sentry/utils/avatar.py:131]: :( lol idk how to type an image, let me try to figure it out

Comment (evanpurkhiser) [reply src/sentry/utils/avatar.py:131]: Look at what data type BytesIO takes

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**